### PR TITLE
Issue LIBCLOUD-354: Add support for volume-related functions to OpenNebula compute driver

### PR DESCRIPTION
**Pull request by ema**

### CODE REVIEW

**Comment by Kami** [comment]
Patch is missing `list_volumes` method (I've just merged this pr into trunk https://github.com/apache/libcloud/pull/118), but besides that it looks good to me.

File: libcloud/test/compute/test_opennebula.py [hunk 1]
```diff
@@ -631,6 +631,52 @@ def test_list_sizes(self):
...
+        nodes = self.driver.list_nodes()
+        node = nodes[0]
+
+        images = self.driver.list_images()
+        image = images[0]
+
+        ret = self.driver.attach_volume(node, image, 'sda')
+        self.assertTrue(ret)
+
+    def test_detach_volume(self):
```
Comment (Kami) [inline libcloud/test/compute/test_opennebula.py:673]: Would be nice to add a test case which xercises a scenario when a node only has a single image associated with it (https://github.com/apache/libcloud/pull/120/files#L0R1174).

File: libcloud/compute/drivers/opennebula.py [hunk 2]
```diff
@@ -868,28 +870,35 @@ def _extract_images(self, compute):
...
-        if len(disks) > 0:
+        # Return all disks when the Node type accepts multiple attached disks
+        # per node.
+        if len(disks) > 1:
+            return disks
+        
+        if len(disks) == 1:
             return disks[0]
-        else:
-            return None
```
Comment (Kami) [inline libcloud/compute/drivers/opennebula.py:892]: FYI - I just spotted while merging merging patch into trunk. I prefer being explicit (Zen of Python agrees with me) so I've changed it to explicitly return "None".

**Comment by Kami** [comment]
I've merged this into trunk, so you can close this pull request now.

Thanks!

File: libcloud/compute/drivers/opennebula.py (hunk 2 above)
Comment (ema) [inline libcloud/compute/drivers/opennebula.py:892]: On 08/07 04:07, Tomaz Muraus wrote:

> > ```
> >          return disks[0]
> > ```
> > -        else:
> > -            return None
> 
> FYI - I just spotted while merging merging patch into trunk. I prefer
> being explicit (Zen of Python agrees with me) so I've changed it to
> explicitly return "None".

Sounds good to me. Thanks for your quick feedback!

Cheers,
    Emanuele

**Pull request closed without merging**